nixos/modules/services/networking/nix-store-gcs-proxy.nix

```diff
@@ -0,0 +1,52 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.nix-store-gcs-proxy;
+  opts = { name, config, ... }: {
+    options = {
+      enable = mkOption {
+        default = true;
+        type = types.bool;
+        description = "Whether to enable proxy for this bucket";
+      };
+      bucketName = mkOption {
+        type = types.str;
+        default = name;
+        description = "Name of Google storage bucket";
+      };
+      address = mkOption {
+        type = types.str;
+      };
+    };
+  };
+  enabledProxies = lib.filterAttrs (n: v: v.enable) cfg;
+  mapProxies = function: lib.mkMerge (lib.mapAttrsToList function enabledProxies);
+in
+{
+  options.services.nix-store-gcs-proxy = mkOption {
+    type = types.attrsOf (types.submodule opts);
+    description = ''
+      Attribute set for a proxy for a particular GCS bucket.
+    '';
+  };
+
+  config.systemd.services = mapProxies (name: cfg: {
+    "nix-store-gcs-proxy-${name}" = {
+      description = "A HTTP nix store that proxies requests to Google Storage";
+      wantedBy = ["multi-user.target"];
+      serviceConfig = {
```

Comment:
Does it have to run as root? Which privileges does this need? Systemd offers a lot of features to reduce the surface of a service.

Reply:
The service is just a process that does everything in memory. It can run as a unprivileged user and doesn't require disk storage.

Reply:
I put it in `systemd.user.services` instead of `systemd.services`. This is how things like redshift are run, let me know if it's not good enough.

Reply:
That is the entirely wrong location. That would add it to every user that logs into the system and not as (one) system service.

A good example is the spacecookie service.

Reply:
@andir Thanks, fixed.